### ISSUE TITLE
datalog crashes when trying to record data more than once per second

### ISSUE THREAD

**Issue by tasota** [CMU-CREATE-Lab/flutter-app-android#127]
Check out this block of code from RecordDataLoggingDialog (commit f1922371d168746c1d2269fa9e3d5236fa75e673)

```
                            // in seconds
                            int interval = 0;

                            String temp = intervalSpinner.getSelectedItem().toString();
                            interval = timeToSeconds(temp);
                            interval = interval / intervalsT;

                            int timePeriodT = Integer.valueOf(timerPeriodString);
                            // in seconds
                            int timePeriod = 0;
                            temp = timePeriodSpinner.getSelectedItem().toString();
                            timePeriod = timeToSeconds(temp);
                            timePeriod = timePeriodT * timePeriod;
                            int sample = timePeriod / interval;
```

**Comment by SteveFulton95** [comment]
@tasota Is the Flutter capable of recording data more than once per second?   If not, should I just check this case and warn the user?

**Comment by tasota** [comment]
@SteveFulton95 yes you're right. We're discussing this now (I haven't looked at the protocol until now that is going on behind the scenes for this).

**Comment by tasota** [comment]
So this might become a design question now for this screen:

https://app.zeplin.io/project.html#pid=57f43918d127f7023581eaab&sid=57f43a09bae86e615c87ff91

Is there another way to phrase this prompt so that the students can only enter values that conform to the protocol?

**Comment by tasota** [comment]
(from the protocol document)

Start logging
Request: ‘l,unix_time,logging_interval,samples’
Notes:
Unix_time is 8 bytes long
Logging_interval is 4 bytes long, and units are seconds
samples is 2 bytes long, and units are in seconds
Example:’l,58064ffc,003c,0a’
Current time is 1476808700, start logging at 60 second intervals, 10 samples

**Comment by SteveFulton95** [comment]
![screenshot_2017-04-07-12-17-53](https://cloud.githubusercontent.com/assets/8007178/24809111/79d06f1a-1b8c-11e7-8ed1-2ba36237b724.png)

What about something like this?

**Comment by mo11yj** [comment]
Yeah, I think adding some verification text like that would work well. If a student enters >60 times per minute we can throw an error, "Please enter 60 or less". I assume we don't have an option for seconds in the drop down? If so we should probably remove that and leave minute and the smallest option.

**Comment by SteveFulton95** [comment]
resolved in commit e9614e353e0e0fe8d66230264d09b8a6ce208932
